src/daq/Notification.cpp

```diff
@@ -42,31 +44,70 @@ void Notification::start() {
     LOG_INF << "Notification stream started";
 }
 
-void Notification::block(Info::MODE mode,
-                         const std::string image_id,
-                         const std::string folder) {
+void Notification::block(Info::MODE mode, const std::string image_id, const std::string folder) {
 
     if (mode != Info::MODE::LIVE) {
         return;
     }
 
-    LOG_DBG << "IMS::Image blocking for image " << image_id;
-    IMS::Image image(*_store, *_stream, TIMEOUT);
-    LOG_DBG << "Acquired image " << image_id;
+    // We make the assumption that we open the stream on start up of the Forwarder.  This
+    // is before getting an endReadout (which calls this method), so we can't be 
+    // "ahead" of reading the data coming from the stream, only behind. Any images that
+    // were on the stream that we skip will be handled by the catchup archiver.
+    //
+    while (1) {
+        //
+        // Call IMS::IMAGE to attempt to set up an image from the stream. If the image 
+        // is null, there were no pending images after TIMEOUT so we thrown an exception.
+        //
+        LOG_DBG << "Trying to read stream to find image " << image_id;
+        IMS::Image image(*_store, *_stream, TIMEOUT);
+        if (!image) {
```

Comment:
```
if (image != nullptr) {
```
probably would be a more "LSST compliant" form for making such tests (see rules: [5-11](https://developer.lsst.io/cpp/style.html?highlight=null#implicit-test-for-0-should-not-be-used-other-than-for-boolean-variables) and [5-34](https://developer.lsst.io/cpp/style.html?highlight=null#nullptr-should-be-used-instead-of-0-and-null)).

Reply:
Fixed